Generative-AI/Youtube_Rag_Chatbot/main.py

```
@@ -0,0 +1,125 @@
+        "instructions": "Go to /docs to try out the API interactively."
+    }
+
+# Endpoint to extract transcript and build vector index
+@app.post("/extract")
+async def extract_transcript(data: ExtractRequest):
+    video_id = data.video_id
+
+    try:
+        transcript = YouTubeTranscriptApi.get_transcript(video_id)
+    except VideoUnavailable:
+        raise HTTPException(status_code=404, detail="Video is unavailable.")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)